variables.tf

```
@@ -1,10 +1,14 @@
-variable "saml_provider_name" {
-  description = "Name of the SAML Identity Provider"
-  type        = string
-}
-
-variable "saml_provider_metadata" {
-  description = "XML metadata for the SAML Identity Provider"
-  default     = ""
-  type        = string
+variable "iam_identity_provider" {
+  description = "Inputs to create a SAML Provider or OpenID Provider"
+  type = object({
+    saml = optional(object({
+      metadata_document = string
+      provider_name     = string
+    }))
+    openid = optional(object({
```

Would prefer to call this attribute `oidc` or `openid_connect`

```suggestion
    oidc = optional(object({
```

---

replaced all `openid` > `openid_connect`